src/pmc_simulation.jl

```diff
@@ -21,42 +21,56 @@ mutable struct PMCSimulation
 end
 
 function _set_up_starting_vectors(
-    start_at, n_replicas, n_spectral, style, initiator, threading, copy_vectors
+    ham, start_at, n_replicas, n_spectral, style, initiator, threading, copy_vectors, minimum_size
 )
     if start_at isa AbstractMatrix && size(start_at) == (n_replicas, n_spectral)
         if eltype(start_at) <: Union{AbstractDVec, RMPI.MPIData} # already dvecs
             if copy_vectors
-                return SMatrix{n_spectral, n_replicas}(deepcopy(v) for v in start_at)
+                return SMatrix{n_replicas, n_spectral}(deepcopy(v) for v in start_at)
             else
-                return SMatrix{n_spectral, n_replicas}(v for v in start_at)
+                return SMatrix{n_replicas, n_spectral}(v for v in start_at)
             end
         else
-            return SMatrix{n_spectral, n_replicas}(
+            return SMatrix{n_replicas, n_spectral}(
                 default_starting_vector(sa; style, initiator, threading) for sa in start_at
             )
         end
+    elseif n_spectral > 1
+        basis = build_basis(ham; minimum_size)
```

Comment:
```suggestion
        basis = build_basis(ham, start_at; minimum_size)
```
As per my earlier comment, we should pass through the information about the starting address.